Kernel/System/DynamicField/Driver/Set.pm

```diff
@@ -130,37 +200,33 @@ sub ValueSet {
     my $DynamicFieldObject = $Kernel::OM->Get('Kernel::System::DynamicField');
     my $BackendObject      = $Kernel::OM->Get('Kernel::System::DynamicField::Backend');
 
-    # loop over the dynamic fields that are included in the Set
-    for my $i ( 0 .. $#{ $Param{DynamicFieldConfig}{Config}{Include} } ) {
-
-        # historical Definitions will be provided by ITSMConfigItems
-        my $Include      = $Param{DynamicFieldConfig}{Config}{Include}[$i];
-        my $DynamicField = $Include->{Definition} // $DynamicFieldObject->DynamicFieldGet(
-            Name => $Include->{DF},
-        );
-
-        if ( !$DynamicField ) {
-            $Kernel::OM->Get('Kernel::System::Log')->Log(
-                Priority => 'error',
-                Message  => "$Param{DynamicFieldConfig}{Name} configured erroneously. 'DF: DynamicFieldName' missing or wrong.",
-            );
-
-            return;
-        }
-
-        # The values for an included dynamic field are the values from the respective column
-        my @FieldValue = map { $_->[$i] } @SetValue;
-
-        if (
-            !$BackendObject->ValueSet(
-                %Param,
-                DynamicFieldConfig => $DynamicField,
-                Value              => \@FieldValue,
-                Set                => 1,
-            )
+    my $Include = $Param{DynamicFieldConfig}{Config}{Include};
+    my $DynamicFieldName = $Param{DynamicFieldConfig}{Name};
+    my $DynamicFields = $GetIncludedDynamicFields->($Include, $DynamicFieldObject, $DynamicFieldName);
+
+    if (!$DynamicFields) {
+        
+        $Success = 0;
+    } else {
+        
+        for my $i (0 .. $#$DynamicFields) {
+
+            my $DynamicField = $DynamicFields->[$i];
+
+            # The values for an included dynamic field are the values from the respective column
+            my @FieldValue = map { $_->[$i] } @SetValue;
+
+            if (
+                !$BackendObject->ValueSet(
+                    %Param,
+                    DynamicFieldConfig => $DynamicField,
+                    Value              => \@FieldValue,
+                    Set                => 1,
+                )
             )
-        {
-            $Success = 0;
+            {
+                $Success = 0;
```

Comment:
The question is whether you even want to try to continue updating the other fields. One maybe could also just return here. Leave a log message though, please. See e.g. https://github.com/RotherOSS/otobo/blob/40a68a17babe7254279c3f27302f037ad8129269/Kernel/System/Ticket.pm#L426-L430
(With an added blank line in front of the return, please.)